test-requirements.in

```diff
@@ -3,26 +3,23 @@ pytest >= 5.0         # for faulthandler in core
 coverage >= 7.2.5
 async_generator >= 1.9
 pyright
-# ipython 7.x is the last major version supporting Python 3.7
-ipython < 7.35        # for the IPython traceback integration tests
+ipython               # for the IPython traceback integration tests
 pyOpenSSL >= 22.0.0   # for the ssl + DTLS tests
 trustme               # for the ssl + DTLS tests
 pylint                # for pylint finding all symbols tests
 jedi                  # for jedi code completion tests
-cryptography>=36.0.0  # 35.0.0 is transitive but fails
+cryptography>=41.0.0  # cryptography<41 segfaults on pypy3.10
 
 # Tools
 black; implementation_name == "cpython"
 mypy; implementation_name == "cpython"
 types-pyOpenSSL; implementation_name == "cpython"
-flake8 < 6.0.0  # 6.0.0 drops python 3.7
+flake8
 flake8-pyproject
 astor          # code generation
 pip-tools >= 6.13.0
 
 # https://github.com/python-trio/trio/pull/654#issuecomment-420518745
-# typed_ast is deprecated as of 3.8, and straight up doesn't compile on 3.10-dev as of 2021-12-13
-typed_ast; implementation_name == "cpython" and python_version < "3.8"
 mypy-extensions; implementation_name == "cpython"
 typing-extensions < 4.7.0
```

Comment:
This should be reverted too. (... I forgot to leave a comment on it. Oops.)

Reply:
Ah, sorry this is pointing towards the typing extensions line!

Reply:
Ah, derp. Done!